### PR TITLE
test: add e2e tests

### PR DESCRIPTION
**Pull request by menduz**

### CODE REVIEW

File: test/components.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+  // start server and other components
+  await startComponents()
+}
+
+// initComponents role is to create BUT NOT START the components,
+// this function is only called once by the Lifecycle manager
+async function initComponents(): Promise<TestComponents> {
+  const logs = createLogComponent()
+
+  const config = createConfigComponent(defaultServerConfig())
```
Comment (menduz) [inline test/components.ts:56]: @gonpombo8 🙌

File: test/components.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
+// this server shuts down after 10100 requests.
```
Comment (menduz) [inline test/components.ts:1]:
```suggestion
```

File: test/ContentClient.E2E.spec.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            creationTimestamp: Date.now()
+          }
+        }
+      })
+    )
+  })
+
+  it('creates the client', async () => {
+    client = new ContentClient({
+      contentUrl: await components.getBaseUrl()
```
Comment (menduz) [inline test/ContentClient.E2E.spec.ts:45]: @gonpombo8 this may be helpful for the `test-helpers`

File: test/ContentClient.E2E.spec.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      }
+    })
+
+    components.router.post(
+      '/entities',
+      multipartParserWrapper(async (ctx) => {
+        expect(ctx.formData.fields).toHaveProperty('entityId')
+        expect(ctx.formData.fields.entityId.value).toEqual('QmENTITY')
+        expect(ctx.formData.files).toHaveProperty('QmA')
+        expect(ctx.formData.files).toHaveProperty('QmB')
```
Comment (menduz) [inline test/ContentClient.E2E.spec.ts:31]:
```suggestion
        expect(ctx.formData.files).toHaveProperty('QmB')
        // TODO: FOR SOME REASON the `deployEntity` _does not_ send the entity itself by default
```

**Review by guidota — APPROVED** [review]

File: test/components.ts (hunk 1 above)
Comment (gonpombo8) [reply test/components.ts:55]: 🎉

**Pull request merged**